python/pyspark/sql/conf.py

```diff
@@ -64,6 +64,96 @@ def _checkType(self, obj, identifier):
                             (identifier, obj, type(obj).__name__))
 
 
+class ConfigEntry(object):
+    """An entry contains all meta information for a configuration"""
+
+    def __init__(self, confKey):
+        """Create a new ConfigEntry with config key"""
+        self.confKey = confKey
+        self.converter = None
+        self.default = _NoValue
+
+    def boolConf(self):
+        """Designate current config entry is boolean config"""
+        self.converter = lambda x: str(x).lower() == "true"
+        return self
+
+    def intConf(self):
+        """Designate current config entry is integer config"""
+        self.converter = lambda x: int(x)
+        return self
+
+    def stringConf(self):
+        """Designate current config entry is string config"""
+        self.converter = lambda x: str(x)
+        return self
+
+    def withDefault(self, default):
+        """Give a default value for current config entry, the default value will be set
+        to _NoValue when its absent"""
+        self.default = default
+        return self
+
+    def read(self, ctx):
+        """Read value from this config entry through sql context"""
+        return self.converter(ctx.getConf(self.confKey, self.default))
+
+class SQLConf(object):
```

Comment:
I think this PySpark SQLConf stuff should be done in a separate Jira/PR.

Reply:
Yea, it should be separate.

Reply:
Yeah, agree, done in #21648.